### ISSUE TITLE
Endless Loop in Select

### ISSUE THREAD

**Issue by akuji1993** [sanusart/react-dropdown-select#55]
When I implement Select something like this: 

```
const [value, setValue] = useState('1');
const onChange = (changedValue) => {
    setValue(changedValue);
}
return <Select values={value} options={['1', '2', '3']} onChange={onChange} />
```

And then pick 2 as the new option, you actually trigger an endless loop. 

This is the case, because in componentDidUpdate in your index.js (Select), you call 

`onChange` with the old state value, then afterwards update your state with the new value given from outside. This means, it will jump from old to new to old to new value (and so on) in our case (with Redux Saga handling some changes asynchronously). It's fixed when you write it like this (which conforms with React concept): 

```
componentDidUpdate() {
...
    this.setState({value: this.props.value}, () => {
         this.props.onChange(this.state.value);
    })
}
```
That way it's using the callback on setState to ensure it only propagates the changed state AFTER it was set on the component state and thus, not propagating the old state. Your order was wrong and even the other way around you cannot rely on setState to be done when onChange is triggered since it works asynchronously.

**Comment by sanusart** [comment]
Thanks @akuji1993 
Published v3.7.0

**Comment by akuji1993** [comment]
Thanks for the quick merge :)
